File: Easy/1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int arr[] = new int[2];
        HashMap<Integer,Integer> hs = new HashMap<>();
        for(int i=0;i<nums.length;i++){
            if(hs.containsKey(target-nums[i])){
                arr[0] = hs.get(target-nums[i]);
                arr[1] = i;
                break;
            }
            else{
                hs.put(nums[i],i);
            }
        }
        return arr;
    }
}

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> map;
        vector<int> ans(2);
        for(int i=0;i<nums.size();i++){
            if(map.find(target-nums[i])!=map.end()){
                ans[0]=map[target-nums[i]];
                ans[1] = i;
            }
            map[nums[i]]=i;
        }
        return ans;
    }
};